### ISSUE TITLE
Add possibility to throttle the instance creation

### ISSUE THREAD

**Issue by winklerm** [jenkinsci/openstack-cloud-plugin#56]
When openstack-cloud-plugin provisions several instances at once, it might make the underlying OpenStack unrealiable (from my experience, when provisioning several (like 5) machines at once, some of them failed to provision). It would be nice to have a configuration option to limit the number of concurrent instances being provisioned.

**Comment by olivergondza** [comment]
I am unsure if this is a problem of openstack itself or of a particular deployment.

**Comment by winklerm** [comment]
When provisioning several (around 10) instances at once, some of the following errors were found in Jenkins log:

```
INFO: Provisioning new openstack node os-rhel7-427
Mar 16, 2016 4:11:08 AM jenkins.plugins.openstack.compute.internal.Openstack throwIfFailed
WARNING: Machine provisioning failed
jenkins.plugins.openstack.compute.internal.Openstack$ActionFailed: Failed to boot server: status=ERROR vmState=error fault=500: Timeout while waiting on RPC response - topic: "network", RPC method: "allocate_for_instance" info: "<unknown>" (null)
        at jenkins.plugins.openstack.compute.internal.Openstack.throwIfFailed(Openstack.java:297)
        at jenkins.plugins.openstack.compute.internal.Openstack.bootAndWaitActive(Openstack.java:189)
        at jenkins.plugins.openstack.compute.JCloudsSlaveTemplate.provision(JCloudsSlaveTemplate.java:241)
        at jenkins.plugins.openstack.compute.JCloudsSlaveTemplate.provisionSlave(JCloudsSlaveTemplate.java:173)
        at jenkins.plugins.openstack.compute.JCloudsCloud$1.call(JCloudsCloud.java:156)
        at jenkins.plugins.openstack.compute.JCloudsCloud$1.call(JCloudsCloud.java:150)
        at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```
```
Mar 16, 2016 4:13:29 AM hudson.slaves.NodeProvisioner update
WARNING: Provisioned slave os-rhel7 failed to launch
java.util.concurrent.ExecutionException: jenkins.plugins.openstack.compute.internal.Openstack$ActionFailed: Failed to boot server: status=ERROR vmState=error fault=500: No valid host was found. Exceeded max scheduling attempts 3 for instance 720d1bc3-d48a-4c9d-999f-61744bed0253 (null)
        at jenkins.plugins.openstack.compute.JCloudsCloud$1.call(JCloudsCloud.java:158)
        at jenkins.plugins.openstack.compute.JCloudsCloud$1.call(JCloudsCloud.java:150)
        at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
        at java.util.concurrent.FutureTask.run(FutureTask.java:262)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
Caused by: jenkins.plugins.openstack.compute.internal.Openstack$ActionFailed: Failed to boot server: status=ERROR vmState=error fault=500: No valid host was found. Exceeded max scheduling attempts 3 for instance 720d1bc3-d48a-4c9d-999f-61744bed0253 (null)
        at jenkins.plugins.openstack.compute.internal.Openstack.throwIfFailed(Openstack.java:297)
        at jenkins.plugins.openstack.compute.internal.Openstack.bootAndWaitActive(Openstack.java:189)
        at jenkins.plugins.openstack.compute.JCloudsSlaveTemplate.provision(JCloudsSlaveTemplate.java:241)
        at jenkins.plugins.openstack.compute.JCloudsSlaveTemplate.provisionSlave(JCloudsSlaveTemplate.java:173)
        at jenkins.plugins.openstack.compute.JCloudsCloud$1.call(JCloudsCloud.java:156)
        ... 6 more
```

**Comment by olivergondza** [comment]
I am closing this for https://github.com/jenkinsci/openstack-cloud-plugin/issues/151.

**Comment by olivergondza** [comment]
Fixed in #368.
